elastic/kibana
# [Discover] Fix index pattern switch behavior

## Summary

This PR takes care of an issue introduced in #92132 when you switch index patterns, found while testing for #88426

* When you were switching from an **index pattern with a timeField** with no columns selected to an **index pattern without a timeField** which contained the same timeField, the sorting by the timefield was preserved, but it was just in the URL, so you had to add the column to remove the sorting (more likely you thought, why does sorting not work ... when trying to sort by another column)
* When you were switching from index pattern with **timeFieldA** to an index pattern with **timeFieldB**, **timeFieldA** was removed, but **timeFieldB** was not applied
* Furthermore it removed another usage of `_source` as default column when switching index patterns. This is now handled in state management

### Testing

* Install `kibana_sample_data_ecommerce` demo data
* Create an index pattern `kibana_sample_data-eco*`, without time field
* Go to Discover
* Select kibana_sample_data_ecommerce
* Switch to `kibana_sample_data-eco*`
* Take a look at the `sort` part of the URL, it should be empty. Before it was containing `order_date,desc`. So it was still sorted, but you had to add the `order_date` column to change this (Or edit the URL) ... confusing

Furthermore
* Switch between index pattern with `timeFieldA` to `timeFieldB` (e.g. from `kibana_sample_data_ecommerce` to `kibana_sample_data_flights`)
* After switching the data table should be sorted by `timeFieldB`



### Checklist
